### PR TITLE
bugfix: convert stf_error to imp_error

### PR DESCRIPTION
**Pull request by jingleizhang**
as the title

### CODE REVIEW

File: tee-worker/app-libs/stf/src/trusted_call.rs [hunk 1]
```diff
@@ -672,6 +667,14 @@ where
...
 }
 
+pub(crate) fn convert_stf_error(e: StfError) -> IMPError {
+	match e {
+		StfError::Dispatch(s) =>
+			IMPError::StfError(ErrorString::truncate_from(s.as_bytes().to_vec())),
+		_ => IMPError::StfError(ErrorString::truncate_from(format!("{:?}", e).as_bytes().to_vec())),
+	}
+}
+
```
Comment (Kailai-Wang) [inline tee-worker/app-libs/stf/src/trusted_call.rs]: This converter is not needed I think.

`StfError` implements `Display` and thus `to_string`, we can simply use `e.to_string()` => for `StfError::Dispatch` too.

I suggest we just implement:
```
impl StfError {
  pub fn to_imp_error(&self) ....
}
```

What do you think?

**Review by Kailai-Wang — APPROVED** [review]
Thanks for spotting this, looks good - I only have one small suggestion.

It is related to https://github.com/litentry/litentry-parachain/issues/1064 -- meaning not all error-handling cases were covered.

We need to have ts-tests too - at least for some of these StfErrors - let's have an issue for that @jingleizhang @Verin1005

**Comment by Kailai-Wang** [comment]
related to #1274 too?

**Comment by jingleizhang** [comment]
a following issue for ts-test is created: https://github.com/litentry/litentry-parachain/issues/1374

File: tee-worker/app-libs/stf/src/trusted_call.rs (hunk 1 above)
Comment (jingleizhang) [reply tee-worker/app-libs/stf/src/trusted_call.rs]: a following issue for ts-test is created: https://github.com/litentry/litentry-parachain/issues/1374

Comment (jingleizhang) [reply tee-worker/app-libs/stf/src/trusted_call.rs]: fixed

**Comment by jingleizhang** [comment]
It is related to https://github.com/litentry/litentry-parachain/issues/1274, but this issue doesn't resolve it completely.

**Pull request merged**